lib/ecto/migration/runner.ex

```diff
@@ -401,4 +413,15 @@ defmodule Ecto.Migration.Runner do
   defp quote_name(prefix, name), do: quote_name(prefix) <> "." <> quote_name(name)
   defp quote_name(name) when is_atom(name), do: quote_name(Atom.to_string(name))
   defp quote_name(name), do: name
+
+  defp reversible_modify?(opts) do
+    opts_keys = opts |> Keyword.delete(:from) |> Keyword.keys() |> Enum.sort()
+
+    case opts[:from] do
+      {_from_type, from_opts} -> opts_keys == from_opts |> Keyword.keys() |> Enum.sort()
+      {from_type} when not is_nil(from_type) -> opts_keys == []
+      from_type when not is_nil(from_type) -> opts_keys == []
```

Comment:
This may be a compatibility break, but the previous behaviour was somewhat faulty. I wonder if we should keep backwards compatibility here?

Reply:
cc @josevalim

Reply:
We should retain the previous behaviour and act differently only if a tuple is given!

Reply:
i'm keeping the previous `from: :atom` opt but not the faulty behaviour
to fix #383, old modify statements that **are** using other options **without** passing their current value to the tuple won't roll back (just like this test: [migration_test.exs#L748-L754](https://github.com/MateusAquino/ecto_sql/blob/master/test/ecto/migration_test.exs#L748-L754))

well, I just added to the previous behaviour a check to find out if the modify is [actually reversible](https://github.com/MateusAquino/ecto_sql/blob/master/lib/ecto/migration/runner.ex#L413)), is this okay? or should we allow them to rollback normally even though it would break the migrations consistency w/o even warning?

Reply:
Unfortunately this will be a breaking change because the previous approach actually worked under some scenarios and users may be relying on those cases. We should keep the previous behaviour as is and instead change the docs to tell users to always prefer the new tuple format!